api/src/domain/service/customer.go

```diff
@@ -0,0 +1,86 @@
+package service
+
+import (
+	"app/domain/errors"
+	"app/domain/model"
+	"app/resources/repository"
+	"bufio"
+	"io"
+	"mime/multipart"
+)
+
+// Customer interface
+type Customer interface {
+	Parse(file multipart.File) (*model.CustomerList, error)
+	read(reader io.Reader) (*model.CustomerList, error)
+	createCustomer(customer *model.Customer) (*model.Customer, error)
+}
+
+// CustomerService struct
+type CustomerService struct {
+	Repository repository.CustomerDB
+}
+
+// Parse recebe um nome de arquivo e retorna o seu conteudo
+func (cs CustomerService) Parse(file multipart.File) (*model.CustomerList, error) {
+
+	defer file.Close()
+	cl, err := cs.read(file)
+
+	if err != nil {
+		return cl, err
+	}
+	return cl, nil
+}
+
+func (cs CustomerService) read(reader io.Reader) (*model.CustomerList, error) {
+
+	var customers []model.Customer
+	cl := &model.CustomerList{}
+	r := bufio.NewReader(reader)
+
+	var errs []error
+	i := 0
+	for ; ;i++{
+		s, _, err := r.ReadLine()
+
+		if err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				return cl, err
+			}
+
+		}
+
+		customer := model.Customer{Name: string(s)}
+		customerDB, err := cs.createCustomer(&customer)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			customers = append(customers, *customerDB)
+		}
+	}
+
+	cl = &model.CustomerList{
+		Data:    customers,
+		Records: int64(len(customers)),
+	}
+
+	if len(errs) == i {
```

Comment:
Eu tenho um pouco de dúvida se é o melhor caso retornar erro de cliente duplicado, não tenho certeza, mas eu retornaria somente um body com a quantidade de clientes inseridos com sucesso, e talvez a quantidade de clientes já existestes, Ex:

`{
	"success": 90,
	"alreadyExist": 10
}`

Oque acha ?

Reply:
Vou criar uma estrutura com esses dois campos. Mas eu ainda retorno o erro caso nenhum cliente é cadastrado?

Reply:
se o success e o alreadyExist for "0" se os dois valores for "0", ai sim mostra erro, pois ele colocou um arquivo vazio ou inválido, acho que assim fica top